nano/node/active_transactions.cpp

```diff
@@ -438,69 +438,70 @@ nano::election_insertion_result nano::active_transactions::insert (std::shared_p
 }
 
 // Validate a vote and apply it to the current election if one exists
-nano::vote_code nano::active_transactions::vote (std::shared_ptr<nano::vote> const & vote_a)
+std::unordered_map<nano::block_hash, nano::vote_code> nano::active_transactions::vote (std::shared_ptr<nano::vote> const & vote)
 {
-	nano::vote_code result{ nano::vote_code::indeterminate };
-	// If all hashes were recently confirmed then it is a replay
-	unsigned recently_confirmed_counter (0);
+	std::unordered_map<nano::block_hash, nano::vote_code> results;
 
 	std::vector<std::pair<std::shared_ptr<nano::election>, nano::block_hash>> process;
 	std::vector<nano::block_hash> inactive; // Hashes that should be added to inactive vote cache
 
 	{
 		nano::unique_lock<nano::mutex> lock{ mutex };
-		for (auto const & hash : vote_a->hashes)
+		for (auto const & hash : vote->hashes)
 		{
-			auto existing (blocks.find (hash));
-			if (existing != blocks.end ())
+			debug_assert (results.find (hash) == results.end ());
```

Comment:
It seems possible the vote could contain the same hash more than once even if the node wouldn't generate this.

Reply:
The processing code is generally resilient to duplicate hashes, do you think this assert should be removed and duplicates ignored?

Reply:
Yea I couldn't find an issue with processing of duplicates other than if someone ran a node with debug asserts enabled it could be triggered to crash with a specifically crafted vote.